state-chain/chains/src/sol/transaction_builder.rs

```diff
@@ -113,18 +114,23 @@ impl SolanaTransactionBuilder {
 				match param.asset {
 					SolAsset::Sol => {
 						compute_limit += COMPUTE_UNITS_PER_FETCH_NATIVE;
+						let fetch_pda_and_bump = derive_fetch_account(
+							param.deposit_fetch_id.address,
+							sol_api_environment.vault_program,
+						)
+						.map_err(SolanaTransactionBuildingError::FailedToDeriveAddress)?;
+
+						// Add extra compute units for each bump derivation required on-chain
+						compute_limit += (AccountBump::MAX - fetch_pda_and_bump.bump) as u32 *
+							COMPUTE_UNITS_PER_BUMP_DERIVATION;
```

Comment:
Seems we could dedup this

Reply:
... for example make it a method on the PdaAndBump struct. `impl PdaAndBump { fn compute_units(&self) -> ComputeUnits }`